### ISSUE TITLE
Compile error

### ISSUE THREAD

**Issue by jjpe** [monto-editor/broker#8]
When I run `cabal update && cabal sandbox init && cabal configure && cabal install --dependencies-only && cabal build`, the result is this compile error:

```
Downloading the latest package list from hackage.haskell.org
Writing a default package environment file to
/Users/j/Documents/TUDelft/MSc/thesis/software/broker/cabal.sandbox.config
Using an existing sandbox located at
/Users/j/Documents/TUDelft/MSc/thesis/software/broker/.cabal-sandbox
Resolving dependencies...
Configuring monto-broker-0.0.1...
Resolving dependencies...
All the requested packages are already installed:
Use --reinstall if you want to reinstall anyway.
Building monto-broker-0.0.1...
Preprocessing library monto-broker-0.0.1...
Preprocessing executable 'broker' for monto-broker-0.0.1...
[1 of 1] Compiling Main             ( broker/Main.hs, dist/build/broker/broker-tmp/Main.o )

broker/Main.hs:57:40: error:
    • Variable not in scope:
        (<>) :: Mod f0 a0 -> Mod f1 a1 -> Mod FlagFields Bool
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*>’ (imported from Options.Applicative),
        ‘*>’ (imported from Options.Applicative)

broker/Main.hs:58:40: error:
    • Variable not in scope:
        (<>) :: Mod f2 a2 -> Mod f3 a3 -> Mod OptionFields String
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*>’ (imported from Options.Applicative),
        ‘*>’ (imported from Options.Applicative)

broker/Main.hs:59:40: error:
    • Variable not in scope:
        (<>) :: Mod f4 a4 -> Mod f5 a5 -> Mod OptionFields String
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*>’ (imported from Options.Applicative),
        ‘*>’ (imported from Options.Applicative)

broker/Main.hs:60:40: error:
    • Variable not in scope:
        (<>) :: Mod f6 a6 -> Mod f7 a7 -> Mod OptionFields String
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*>’ (imported from Options.Applicative),
        ‘*>’ (imported from Options.Applicative)

broker/Main.hs:61:40: error:
    • Variable not in scope:
        (<>) :: Mod f8 a8 -> Mod f9 a9 -> Mod OptionFields Port
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*>’ (imported from Options.Applicative),
        ‘*>’ (imported from Options.Applicative)

broker/Main.hs:62:40: error:
    • Variable not in scope:
        (<>) :: Mod f10 a10 -> Mod f11 a11 -> Mod OptionFields Port
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*>’ (imported from Options.Applicative),
        ‘*>’ (imported from Options.Applicative)

broker/Main.hs:68:5: error:
    • Variable not in scope:
        (<>) :: InfoMod a12 -> InfoMod a13 -> InfoMod Options
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*>’ (imported from Options.Applicative),
        ‘*>’ (imported from Options.Applicative)
```

Some background info:
- I'm using OS X, in case that can matter in the Haskell world
- `ghc --version` yields
  `
  The Glorious Glasgow Haskell Compilation System, version 8.0.1
  `

**Comment by StareInTheAir** [comment]
I think you are on the master branch. All development currently happens on the file-dependencies branch. Could you try on that branch again, if there is no concrete reason to be on master? On file-dependencies I added convenience shell scripts for building and running the broker with either cabal or stack.

**Comment by svenkeidel** [comment]
@StareInTheAir, nonetheless the master branch should be stable and should not produce compile errors. So either try to fix the issue in master or merge the feature branch.

**Comment by svenkeidel** [comment]
The issue was caused by the library _optparse-applicative_ that removed the export of the _mappend_ operator `(<>)` from the module `Options.Applicative` in their most recent release 0.13.0.0. My suggestion would be to import `(<>)` from `Data.Monoid` directly and to import `Options.Applicative` qualified. This way master should be compatible with older and the most recent version of _optparse-applicative_

**Comment by jjpe** [comment]
Indeed I was on `master`, and the `file-dependencies` branch compiles normally, so for now that fixes it.

**Comment by StareInTheAir** [comment]
This compile error should also be present on file-dependencies, when using cabal. The currently latest stack resolver (lts-6.13) doesn't yet use optparse-applicative 0.13.0, so stack should still work. @jjpe, did you use cabal or stack for building? I don't have have cabal installed atm, so I can't test it.

**Comment by jjpe** [comment]
I'm using `cabal`, and the most recent commit in the log for the `file-dependencies` branch is `edac019d518b048080b7516bcf829d69f3871f44`.

**Comment by StareInTheAir** [comment]
I did some further testing. For me the error also happened on file-dependencies when using cabal. As suggested by the opt-parse authors and Sven, importing `(<>)` from `Data.Monoid` fixes the error (6f6174e9418c36b5e1ef54a1bdc65c32f5aff2b1), but raises a warning, when using optparse-applicative 0.12 (which is the version stack still uses). The broker runs fine, despite the warning. Once the stack resolver updates, the warning will be gone to.

I also replaced the `cabal` build instructions with `stack` (54bd54a0be9e963feea0140c603a5a75176836da). You can still use `cabal`, but I would suggest switching to `stack` (easier to use and more robust).
